### ISSUE TITLE
Backport #6925

### ISSUE THREAD

**Issue by shaneutt** [terraform-providers/terraform-provider-azurerm#8436]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Backport https://github.com/terraform-providers/terraform-provider-azurerm/pull/6925 to `v1` to fix timing issues that can occur for users still on `v1`.

### New or Affected Resource(s)

* azurerm_role_assignment

### References

* https://github.com/terraform-providers/terraform-provider-azurerm/pull/6925

**Comment by tombuildsstuff** [comment]
hi @shaneutt 

Thanks for opening this issue.

As you've mentioned #6925 has fixed this bug in version 2.10 - unfortunately we have no plans to backport this to the 1.x branch. However you should be able to upgrade to version 2.x (the latest at the time of writing being `v2.27.0`) of the Azure Provider and get this fix that way, via updating the provider block:

```
provider "azurerm" {
  version = "= 2.27.0"
  features {}
}
```

and then running `terraform init` (having [worked through the upgrade guide](https://www.terraform.io/docs/providers/azurerm/guides/2.0-upgrade-guide.html)) which should allow you to access this fix.

Since we have no plans to do any further 1.x releases, unfortunately we have no plans to backport this and as such I'm going to close this issue for the moment.

Thanks!

**Comment by shaneutt** [comment]
I appreciate the quick response so I can set expectations back at my org :+1:
